File: Tema 06 - React/petshelter.ui/src/Models/Pet.ts
import type { Person } from "./Person";

export class Pet {
    id?: number;
    name: string;
    imageUrl: string;
    description: string;
    birthdate: Date;
    type: string;
    rescuer?: Person;
    adopter?: Person;
    weightInKg: number

    constructor(name: string, imageUrl: string, type: string, description:string, birthdate: Date, weightInKg: number, id?: number, rescuer?: Person, adopter?: Person) {
        this.id = id;
        this.name = name;
        this.imageUrl=imageUrl;
        this.type = type;
        this.description = description;
        this.birthdate = birthdate;
        this.rescuer = rescuer;
        this.adopter = adopter;
        this.weightInKg = weightInKg;
    }
}

File: Tema 07 - Clean Code/After/petshelter.api/PetShelter.DataAccessLayer/Models/Pet.cs
namespace PetShelter.DataAccessLayer.Models;

public class Pet : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string ImageUrl { get; set; }
    public string Type { get; set; }
    public string? Description { get; set; }
    public DateTime Birthdate { get; set; }
    public bool IsHealthy { get; set; }
    public decimal WeightInKg { get; set; }
    public bool IsSheltered { get; set; }
    public int? RescuerId { get; set; }
    public int? AdopterId { get; set; }

    public Person? Adopter { get; set; }
    public Person Rescuer { get; set; }
}
